### PR TITLE
add documentation for "match-features"

### PR DESCRIPTION
**Pull request by arnej27959**
* also be more explicit about how "summary-features" works
* shuffle some ordering to keep related concepts closer to each other

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@havardpe for discussion
@kkraune FYI

### CODE REVIEW

File: en/reference/schema-reference.html [hunk 1]
```diff
@@ -720,17 +721,21 @@ <h2 id="rank-profile">rank-profile</h2>
...
-  <a href="query-api-reference.html#ranking.listFeatures">rankfeatures</a>.</td>
-<td>Zero or more</td>
-</tr>
 <tr><td><a href="#secondphase-rank">second-phase</a></td>
 <td>The ranking config to be used for second-phase ranking.</td>
 <td>Zero or one</td>
 </tr>
 <tr><td><a href="#summary-features">summary-features</a></td>
-<td>The <a href="../reference/rank-features.html">rank features</a> to be returned with each hit.</td>
+<td>The <a href="../reference/rank-features.html">rank features</a> to be returned with each hit, computed n the <em>fill</em> phase.</td>
```
Comment (kkraune) [inline en/reference/schema-reference.html:729]:
```suggestion
<td>The <a href="../reference/rank-features.html">rank features</a> to be returned with each hit, computed in the <em>fill</em> phase.</td>
```

File: en/reference/schema-reference.html [hunk 2]
```diff
@@ -720,17 +721,21 @@ <h2 id="rank-profile">rank-profile</h2>
...
+<td>Zero or more</td>
+</tr>
+<tr><td><a href="#match-features">match-features</a></td>
+<td>The <a href="../reference/rank-features.html">rank features</a> to be returned with each hit, computed n the <em>match</em> phase.</td>
```
Comment (kkraune) [inline en/reference/schema-reference.html:733]:
```suggestion
<td>The <a href="../reference/rank-features.html">rank features</a> to be returned with each hit, computed in the <em>match</em> phase.</td>
```

File: en/reference/schema-reference.html [hunk 3]
```diff
@@ -1095,11 +1100,16 @@ <h2 id="secondphase-rank">second-phase</h2>
...
+  List of <a href="../reference/rank-features.html">rank features</a> to be included with each hit
+  (in a field with the name <code>summaryfeatures</code> - see also
+  <a href="../ranking-expressions-features.html#accessing-feature-function-values-in-results">feature
+  values in results</a> documentation).
   If not specified, the summary features are as specified in the parent profile (if any).
   To inherit the summary features from the parent profile <em>and</em> specify additional features,
   specify explicitly that the summary features should be inherited from the parent as shown below.
   Refer to the <a href="../schema-inheritance.html">schema inheritance guide</a> for examples.
+  The rank features specified here are computed in the <em>fill</em> phase
+  (see <a href="searcher-development.html#multiphase-searching">multiphase searching</a>).
```
Comment (kkraune) [inline en/reference/schema-reference.html:1112]:
```suggestion
  (see <a href="../searcher-development.html#multiphase-searching">multiphase searching</a>).
```

**Review by kkraune — APPROVED** [review]
This is great!

**Review by havardpe — APPROVED** [review]

**Pull request merged**